From 9f86a39fc37325cd515c3a175e74e76f021c0460 Mon Sep 17 00:00:00 2001
From: Iain Sandoe <iain@sandoe.co.uk>
Date: Wed, 17 Apr 2019 22:12:32 +0530
Subject: [PATCH] [llvm] Add support for force_cpusubtype_ALL and -n.

We want to be able to make a look-alike for cctools as.
---
 llvm/include/llvm/Target/TargetOptions.h | 10 +++++++++-
 1 file changed, 9 insertions(+), 1 deletion(-)

diff --git a/llvm/include/llvm/Target/TargetOptions.h b/llvm/include/llvm/Target/TargetOptions.h
index 07ed773de55..26c7df2e326 100644
--- a/llvm/include/llvm/Target/TargetOptions.h
+++ b/llvm/include/llvm/Target/TargetOptions.h
@@ -111,7 +111,8 @@ namespace llvm {
           NoTrapAfterNoreturn(false), EmulatedTLS(false),
           ExplicitEmulatedTLS(false), EnableIPRA(false),
           EmitStackSizeSection(false), EnableMachineOutliner(false),
-          SupportsDefaultOutlining(false), EmitAddrsig(false) {}
+          SupportsDefaultOutlining(false), EmitAddrsig(false),
+          NoInitialTextSect(false), CPUSubTypeALL(false) {}
 
     /// PrintMachineCode - This flag is enabled when the -print-machineinstrs
     /// option is specified on the command line, and should enable debugging
@@ -241,6 +242,13 @@ namespace llvm {
     /// Emit address-significance table.
     unsigned EmitAddrsig : 1;
 
+    /// Don't assume that the object starts with a .text section.
+    unsigned NoInitialTextSect : 1;
+
+    /// Accept and produce code for any mnemonic in the Arch (even if it's not
+    /// supported by the main mcpu defined).
+    unsigned CPUSubTypeALL : 1;
+
     /// FloatABIType - This setting is set by -float-abi=xxx option is specfied
     /// on the command line. This setting may either be Default, Soft, or Hard.
     /// Default selects the target's default behavior. Soft selects the ABI for
